text/0000-crt-static.md

```diff
@@ -0,0 +1,371 @@
+solving this problem.
+
+### References
+
+- [Issue about MSVCRT static linking]
+  (https://github.com/rust-lang/libc/issues/290)
+- [Issue about musl dynamic linking]
+  (https://github.com/rust-lang/rust/issues/34987)
+- [Discussion on issues around glgobal codegen configuration]
+  (https://internals.rust-lang.org/t/pre-rfc-a-vision-for-platform-architecture-configuration-specific-apis/3502)
+- [std-aware Cargo RFC]
+  (https://github.com/rust-lang/libc/issues/290).
+  A proposal to teach Cargo to build the standard library. Rebuilding of std will
+  likely in the future be influenced by `-C target-feature`.
+- [Cargo's documentation on build-script environment variables]
+  (https://github.com/rust-lang/libc/issues/290)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* Working with `RUSTFLAGS` can be cumbersome, but as explained above it's
+  planned that eventually there's a much more ergonomic configuration method for
+  other codegen options like `target-cpu` which would also encompass the linkage
+  of the CRT.
+
+* Adding a feature which is intended to never be stable (`#[link(.., cfg(..))]`)
+  is somewhat unfortunate but allows sidestepping some of the more thorny
+  questions with how this works. The stable *semantics* will be that for some
+  targets the `--cfg crt_link=...` directive affects the linkage of the CRT,
+  which seems like a worthy goal regardless.
+
+* The lazy semantics of `#[link(cfg(..))]` are not so obvious from the name (no
+  other `cfg` attribute is treated this way). But this seems a minor issue since
+  the feature serves one implementation-specif purpose and isn't intended for
+  stabilization.
+
+# Alternatives
+[alternatives]: #alternatives
+
+* One alternative is to add entirely new targets, for example
+  `x86_64-pc-windows-msvc-static`. Unfortunately though we don't have a great
+  naming convention for this, and it also isn't extensible to other codegen
+  options like `target-cpu`. Additionally, adding a new target is a pretty
+  heavyweight solution as we'd have to start distributing new artifacts and
+  such.
+
+* Another possibility would be to start storing metadata in the "target name"
+  along the lines of `x86_64-pc-windows-msvc+static`. This is a pretty big
+  design space, though, which may not play well with Cargo and build scripts, so
+  for now it's preferred to avoid this rabbit hole of design if possible.
+
+* Finally, the compiler could simply have an environment variable which
+  indicates the CRT linkage. This would then be read by the compiler and by
+  build scripts, and the compiler would have its own back channel for changing
+  the linkage of the C library along the lines of `#[link(.., cfg(..))]` above.
+
+* Another approach has [been proposed recently][rfc-1684] that has
+  rustc define an environment variable to specify the C runtime kind.
+
+[rfc-1684]: https://github.com/rust-lang/rfcs/pull/1684
+
+* Instead of extending the semantics of `-C target-feature` beyond "CPU
```

Comment:
:+1: I would prefer to have two namespaces for this. `CARGO_CFG_TARGET_FEATURE` for the existing LLVM codegen features and another one for Rust specific stuff like `crt-static`. It eliminates the possibility of a new (LLVM) target-feature that collides, in name, with a Rust specific feature like crt-static (assuming we'll grow more of those than just crt-static).